job-server/src/main/scala/spark/jobserver/JobManagerActor.scala

```diff
@@ -209,6 +214,27 @@ class JobManagerActor(contextConfig: Config, daoActor: ActorRef)
         }
       }
     }
+    case GetSparkWebUIUrl => {
+      if (jobContext.sparkContext == null) {
```

Comment:
This keeps it consistent with the checks in the rest of the file. Also, I am fairly sure that due to the way the jobContext controls the lifetime of sparkContext, `isStopped` would result in a NPE. Might be a bigger refactor to make that doable